frontend/src/components/core/MainMenu/MainMenu.test.tsx

```diff
@@ -86,130 +90,63 @@ describe("App", () => {
     const props = getProps({
       hostMenuItems: items,
     })
-    const wrapper = mount(<MainMenu {...props} />)
-    const popoverContent = wrapper.find("StatefulPopover").prop("content")
-
-    // @ts-ignore
-    const menuWrapper = mount(popoverContent(() => {}))
-
-    // @ts-ignore
-    const menuLabels = menuWrapper
-      .find("MenuStatefulContainer")
-      .at(0)
-      .prop("items")
-      // @ts-ignore
-      .map(item => item.label)
-    expect(menuLabels).toEqual([
-      "Rerun",
-      "Settings",
-      "Print",
-      "Record a screencast",
-      "Report a bug",
-      "Get help",
-      "View app source",
-      "Report bug with app",
-      "About",
-    ])
-
-    // @ts-ignore
-    const devMenuLabels = menuWrapper
-      .find("MenuStatefulContainer")
-      .at(1)
-      .prop("items")
-      // @ts-ignore
-      .map(item => item.label)
-    expect(devMenuLabels).toEqual([
-      "Developer options",
-      "Clear cache",
-      "Streamlit Cloud",
-      "Report a Streamlit bug",
-      "Visit Streamlit docs",
-      "Visit Streamlit forums",
-    ])
+    const wrapper = render(<MainMenu {...props} />)
+    await user.click(screen.getByRole("button"))
+
+    await waitFor(() =>
+      expect(
+        wrapper
+          .getAllByRole("option")
+          .map(item => item.querySelector("span:first-of-type")?.textContent)
+      ).toEqual([
+        "Rerun",
+        "Settings",
+        "Print",
+        "Record a screencast",
+        "View app source",
+        "Report bug with app",
+        "About",
+        "Developer options",
+        "Clear cache",
+      ])
+    )
   })
 
-  it("should render core set of menu elements", () => {
+  it("should render core set of menu elements", async () => {
+    const user = userEvent.setup()
```

Comment:
Nit: Do we really need this? Reading the documentation of this method, it seems like this method is for:
`This allows to write multiple consecutive interactions that behave just like the described interactions by a real user.` which doesn't seem to be needed because there's only 1 click

Reply:
Can you suggest what to write instead? I'm creating this variable here because `user.click` is the only way I know in my 3 hours experience with the React Testing Library to fire the click event. And without it, the menu won't expand.

Reply:
Actually I just readup on the documentation of userEvent and it seems like they use it. sorry! Will resolve this and this seems like the right thing to do. 

Thanks for converting!